net-core/Ical.Net/Ical.Net/Components/UniqueComponent.cs

```diff
@@ -96,8 +96,28 @@ protected override void OnDeserialized(StreamingContext context)
         public int CompareTo(UniqueComponent other)
             => string.Compare(Uid, other.Uid, StringComparison.OrdinalIgnoreCase);
 
+        public bool Equals(UniqueComponent other)
+        {
+            // TODO: Casting UniqueComponent to RecurringComponent, i.e. a subclass, creates tight coupling of the two classes. 
+            //      This implementation should be moved to RecurringComponent class instead. 
+
+            if (other is RecurringComponent && other != this)
+            {
+                var r = (RecurringComponent)other;
+                if (Uid != null)
+                {
+                    return Uid.Equals(r.Uid);
+                }
+                return Uid == r.Uid;
+            }
```

Comment:
* Using the `Uid` as the basis for equality isn't consistent with the rest of the library.
* This method should delegate to RecurringComponent.Equals() in the event that the element is a recurring, and call its base class in the event that the element is NOT recurring. And RecurringComponent should know how to do a reasonable comparison and hash.

I think `RecurringComponent` is missing the `IEquatable<T>`, and isn't defining an `Equals` method, so we have nothing to delegate to.

If you want, you can add the `IEquatable<T>` to the `RecurringComponent`, and define an Equals function. I think it will require some thought for both `GetHashCode` and `Equals`. If you don't want to do that, or would prefer to do it as a separate pull request, you can remove the `IEquatable` from this class, and add it back in with the changes to RecurringComponent.